src/Autofac/Features/Scanning/ScanningRegistrationExtensions.cs

```diff
@@ -86,13 +86,11 @@ private static void ScanTypes(IEnumerable<Type> types, IComponentRegistryBuilder
             rb.RegistrationData.Services.OfType<IServiceWithType>().All(swt =>
                 swt.ServiceType.IsAssignableFrom(t)));
 
-        Func<Type, IRegistrationBuilder<object, ConcreteReflectionActivatorData, SingleRegistrationStyle>> scannedConstructor =
-            (type) => RegistrationBuilder.ForType(type);
+        IRegistrationBuilder<object, ConcreteReflectionActivatorData, SingleRegistrationStyle> TypeBuilderFactory(Type type) => RegistrationBuilder.ForType(type);
 
-        Action<IComponentRegistryBuilder, IRegistrationBuilder<object, ConcreteReflectionActivatorData, SingleRegistrationStyle>> register =
-            (cr, scanned) => RegistrationBuilder.RegisterSingleComponent(cr, scanned);
+        void SingleComponentRegistration(IComponentRegistryBuilder cr, IRegistrationBuilder<object, ConcreteReflectionActivatorData, SingleRegistrationStyle> scanned) => RegistrationBuilder.RegisterSingleComponent(cr, scanned);
```

Comment:
Can these be declared as static local methods?

Reply:
Oh, I see what happened - the analyzer considered `cr` in the body of the function to be a closure over the `ScanTypes` _parameter_ rather than its own local parameter of the same name. I can mark it static, and I'll update the parameter names to disambiguate.